stl/src/vector_algorithms.cpp

```diff
@@ -635,13 +696,103 @@ namespace {
             return _mm_max_epu8(_First, _Second);
         }
 
-        static __m128i _Mask_cast(__m128i _Mask) noexcept {
+        static __m128i _Mask_cast(const __m128i _Mask) noexcept {
             return _Mask;
         }
-#endif // !_M_ARM64EC
     };
 
-    struct _Minmax_traits_2 {
+    struct _Minmax_traits_1_avx : _Minmax_traits_1_base, _Minmax_traits_avx_base {
+        static __m256i _Load(const void* _Src) noexcept {
+            return _mm256_loadu_si256(reinterpret_cast<const __m256i*>(_Src));
+        }
+
+        static __m256i _Sign_correction(const __m256i _Val, const bool _Sign) noexcept {
+            alignas(32) static constexpr _Unsigned_t _Sign_corrections[2][32] = {
+                {0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80,
+                    0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80},
+                {}};
+            return _mm256_sub_epi8(_Val, _mm256_load_si256(reinterpret_cast<const __m256i*>(_Sign_corrections[_Sign])));
+        }
+
+        static __m256i _Inc(const __m256i _Idx) noexcept {
+            return _mm256_add_epi8(_Idx, _mm256_set1_epi8(1));
+        }
+
+        template <class _Fn>
+        static __m256i _H_func(const __m256i _Cur, _Fn _Funct) noexcept {
+            const __m128i _Shuf_bytes = _mm_set_epi8(14, 15, 12, 13, 10, 11, 8, 9, 6, 7, 4, 5, 2, 3, 0, 1);
+            const __m128i _Shuf_words = _mm_set_epi8(13, 12, 15, 14, 9, 8, 11, 10, 5, 4, 7, 6, 1, 0, 3, 2);
+
+            __m256i _H_min_val = _Cur;
+            _H_min_val         = _Funct(_H_min_val, _mm256_permutex_epi64(_H_min_val, _MM_SHUFFLE(1, 0, 3, 2)));

```

Comment:
The same [vpermq](https://www.felixcloutier.com/x86/vpermq) instruction with the same description may be encoded differently.

Quite possible that the compiler may use the VEX encoding rather than EVEX for `_mm256_permutex_epi64`.

Still it should be replaced with `_mm256_permute4x64_epi64`

Reply:
As for why `_mm256_permutex_epi64` even exists. There are masked flavors and the EVEX instruction is always masked. So they added masked intrinsics, and redundant unmasked intrisic, as usually, probably for consistency with the cases, where they added totally new operations with and without a mask, see [_permutex_epi64](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_permutex_epi64&ig_expand=4957) search on Intel Intrinsics Guide.

Some compiler even choice not to implement some of such redundant intrinsics, see [this Q&A](https://stackoverflow.com/q/53604986/2945027) as one of the examples.